docs/source/onnx/hotwords/index.rst

```diff
@@ -0,0 +1,919 @@
+
+      .. caution::
+
+        The hotwords in hotwords-file should be tokenized to modeling units (i.e
+        the symbols in tokens.txt
+
+
+    For models trained on ``cjkchar`` it looks like:
+
+     .. code-block::
+
+       语 音 识 别
+       深 度 学 习
+
+    For models trained on ``bpe`` (English like languages) it looks like:
+
+     .. code-block::
+
+      ▁SP E E CH ▁RE C O G N ITION
+      ▁DE E P ▁LE AR N ING
+
+    For models trained on ``cjkchar+bpe`` it looks like:
+
+     .. code-block::
+
+      ▁SP E E CH 识 别
+      ▁SP E E CH ▁RE CO G N ITION    
+      深 度 学 习
+
+  - ``hotwords-score``
+
+    The boosting score for each matched token.
+
+     .. note::
+
+       We match the hotwords at token level, so the ``hotwords-score`` is applied
+       at token level.
+
+
+We provide a command line tool to convert hotwords to tokens, you can see the
+usage as follows:
+
+.. code-block::
+
+   sherpa-onnx text2token --help
+   Usage: sherpa-onnx text2token [OPTIONS] INPUT OUTPUT
+   
+   Options:
+     --tokens TEXT       The path to tokens.txt.
+     --tokens-type TEXT  The type of modeling units, should be cjkchar, bpe or
+                         cjkchar+bpe
+     --bpe-model TEXT    The path to bpe.model.
+     --help              Show this message and exit.
+
+The tool has three options:
+
+  - ``tokens``
+
+    The file path of the tokens.txt, you can find tokens.txt in any of our
+    pretrained model.
+
+  - ``tokens-type``
+
+    The modeling unit used to train the models. Three kinds
+    of ``tokens_type`` are supported now, ``cjkchar``, ``bpe`` and ``cjkchar+bpe``.
+    The ``tokens-type`` tells the systems how to encode ``hotwords`` into tokens.
+
+  - ``bpe-model``
+
+    The file path of the bpe model used to generate the ``tokens.txt``, it is also
+    used to encode the ``hotwords``.
+    Only used when ``tokens-type`` is ``bpe`` or ``cjkchar+bpe``.
+
+
+The main difference of using hotwords feature is about the modeling units (i.e. tokens_type).
+The following shows how to use it for different modeling units.
+
+.. hint::
+
+   You can use any transducer models here `<https://k2-fsa.github.io/sherpa/onnx/pretrained_models/index.html>`_,
+   we just choose three of them randomly for the following examples.
+
+
+.. note::
+
+   In the following example, we use a non-streaming model, if you are using a
+   streaming model, you should use ``sherpa-onnx``. ``sherpa-onnx-alsa``,
+   ``sherpa-onnx-microphone``, ``sherpa-onnx-microphone-offline``,
+   ``sherpa-onnx-online-websocket-server`` and ``sherpa-onnx-offline-websocket-server``
+   all support hotwords.
+
+
+Modeling unit is bpe
+^^^^^^^^^^^^^^^^^^^^
+
+**Download the model**
+
+.. code-block:: bash
+
+   cd /path/to/sherpa-onnx
+
+   GIT_LFS_SKIP_SMUDGE=1 git clone https://huggingface.co/csukuangfj/sherpa-onnx-zipformer-en-2023-04-01
+   cd sherpa-onnx-zipformer-en-2023-04-01
+   git lfs pull --include "*.onnx"
+   git lfs pull --include "bpe.model"
```

Comment:
could you upload `bpe.model` to this repo?

Reply:
I think I already made a pull request

Reply:
<img width="1043" alt="Screenshot 2023-09-13 at 10 22 51" src="https://github.com/k2-fsa/sherpa/assets/5284924/85469698-1982-47a2-9280-0cbc04aeaa4b">

There are no pull-requests at all.

Reply:
Ah, I think I made one here (https://huggingface.co/csukuangfj/sherpa-onnx-streaming-zipformer-bilingual-zh-en-2023-02-20) but not https://huggingface.co/csukuangfj/sherpa-onnx-zipformer-en-2023-04-01, will do it when huggingface is available.